library/core/src/str/pattern.rs

```diff
@@ -1149,15 +1150,16 @@ unsafe impl<'a, 'b> Searcher<'a> for StrSearcher<'a, 'b> {
                     self.needle.as_bytes(),
                     is_long,
                 ) {
-                    SearchStep::Reject(a, mut b) => {
+                    byte_pattern::SearchStep::Reject(a, mut b) => {
                         // skip to next char boundary
                         while !self.haystack.is_char_boundary(b) {
                             b += 1;
                         }
                         searcher.position = cmp::max(b, searcher.position);
                         SearchStep::Reject(a, b)
                     }
-                    otherwise => otherwise,
+                    byte_pattern::SearchStep::Match(a, b) => SearchStep::Match(a, b),
+                    byte_pattern::SearchStep::Done => SearchStep::Done,
```

Comment:
I duplicated `SearchStep` because a public type and its documentation refers to the `Searcher` trait. The `byte_pattern` module will have it's own `Searcher` trait (or `ByteSearcher` maybe) and so that documentation would be misleading one way or the other.